client/src/components/river/index.tsx

```diff
@@ -155,11 +206,60 @@ const FhirRiverView = (): React.ReactElement => {
             disabled={!selectedSource.id || credentialsMissing}
             loading={running}
             className="button-submit"
-            onClick={onClickRun}
+            onClick={onClickCreateBatch}
           >
             Make the river flow
           </Button>
         </div>
+        <h1>Cancel a batch</h1>
+        <StringSelect
+          items={
+            batchList.data
+              ? Object.keys(batchList.data).map(
+                  (batchId: string) => batchList[batchId].timestamp
+                )
+              : []
+          }
+          inputItem={
+            !!selectedBatch && !batchListError
+              ? batchList[selectedBatch].timestamp
+              : 'Select a batch to cancel'
+          }
+          onChange={(item: string): void => {
+            const batchId = Object.keys(batchList.data).find(
+              batchId => batchList[batchId].timestamp === item
+            );
+            if (batchId) setSelectedBatch(batchId);
+          }}
+        />
+        <ul>
```

Comment:
This is a list of the resources processed in the selected batch, right?
Could you add a small paragraph or find another way to make that clear to the user?

Reply:
Oh right, maybe it's the selector's job to do that. What do you think?